### DISCUSSION TITLE
Questions about the dependence of elastic tensor on phase field variable

### DISCUSSION THREAD

**Discussion by ZhigangPu** [idaholab/moose#19053] [Q&A Modules: Phase field] [answered]
Dear,
   I am new to the moose and right now I am trying to couple the phase field and tensor mechanics to simlulate dendrite growth in elastic field. I am reading examples in the  `modules/phase_field-mechanics/combined` directory. **I am confused about the way Moose specify the elastic tensor's dependence and strain's dependence on phase field varible**. 

For example, in the input file `Conserved.i`,
For elastic tensor, no field is reserved for the choice of phase variable in following section, but indeed elastic tensor is different across different phase field.
![image](https://user-images.githubusercontent.com/24690469/136791638-57d0aa5e-c503-4957-80ec-2329c3cdfcd5.png)

For strain and elastic energy, `args` are specified as `c` in the correponding section, but I cannot understand the way `c` interacts with strain and energy. I've read some official document on `ComputeVariableEigenstrain` and `ElasticEnergyMaterial`, but there seems to be no direct explanation.
![image](https://user-images.githubusercontent.com/24690469/136792002-b7b27bb3-c5f4-4d16-ab8e-c4608cc7e868.png)

Any reponse will be highly appreciated!

Best,
Zhigang

**Comment by hugary1995** [comment, marked as answer]
Just by reading that input file it seems to me that
- the eigenstrain is defined as `0.1 * c * I`, where `I` is the second order identity tensor;
- ~the strain purely depends on the eigenstrain, hence the strain is defined as `-0.1 * c * I`;~ Nope, there's also elastic deformation.
- the elasticity tensor is homogeneous (doesn't depend on `c`).

**Reply by hugary1995** [reply]
BTW, I'm not a fan of coupling "Eulerian" dendritic growth with "Lagrangian" mechanics. It's not as simple as it seems to be.

**Reply by ZhigangPu** [reply]
Dear hugary:

Thanks for the comments. Yeah, but it's wired the elasticity tensor is homogeneous since there are two saperate phases. As for the difficulty of coupling between Eulerian method and Lagrangian method, thanks for pointing out and I will think more about it.

Best
zhigang

**Reply by hugary1995** [reply]
I think that's a perfectly valid example for demonstration purposes. If you want to have variable dependent elasticity tensor you can use `ComputeVariableElasticityTensor` or something like that, I don't remember the exact name.

**Reply by ZhigangPu** [reply]
Dear hugary:

Great! Then I'll try the `ComputeVariableElasticityTensor` object.
Many thanks for your kindly remind! I really need this.

Best,
zhigang

**Comment by biaogxb** [comment]
Hello, can I have your wechat or Facebook? I'm also using moose to do dendrite problems. Can I have an in-depth exchange with you？looking forward to your reply.
my emial ：709806584@qq.com
